### ISSUE TITLE
KS version not respected

### ISSUE THREAD

**Issue by alexiri** [pykickstart/pykickstart#444]
Koji uses pykickstart to read a kickstart template, modify it and write out a new kickstart. pykickstart isn't respecting the version of the original template and it's regenerating a new kickstart that does not comply with the version that was requested.

I wrote a test case to (hopefully) illustrate the error:

```python
from pykickstart.version import RHEL8
class Packages_Options_VersionRespected(ParserTest):
    def __init__(self, *args, **kwargs):
        ParserTest.__init__(self, *args, **kwargs)
        self.ks = """
#version=RHEL8
%packages --instLangs="es_ES" --excludeWeakdeps
%end
"""

    def runTest(self):
        self.parser.readKickstartFromString(self.ks)

        # Verify that the options are parsed as expected
        self.assertTrue(self.handler.packages.excludeWeakdeps)
        self.assertEqual(self.handler.packages.instLangs, "es_ES")

        # And then test that all those options would be printed out
        # correctly.
        self.assertEqual(str(self.handler.version), RHEL8)
        self.assertEqual(str(self.handler.packages), "\n%packages --instLangs=es_ES --excludeWeakdeps\n\n%end\n")
```

The first two asserts pass as expected, but the last two fail. The generated kickstart looks like this:

```
# Generated by pykickstart v3.44
#version=DEVEL
# System bootloader configuration
bootloader --location=none
  
%packages --inst-langs=es_ES --exclude-weakdeps
%end
```

Of course, when this gets passed to a RHEL8 version of anaconda, it complains because `--inst-langs` and `--exclude-weakdeps` are unrecognized arguments.

Note: in my current Koji installation (which uses pykickstart v3.32), the `version` is generated correctly but the package arguments are still wrong. This may just be that my test is wrong, I'm not sure.

**Comment by alexiri** [comment]
Digging into this further, here's a minimal example of what Koji is doing and how pykickstart's output is invalid for the requested version:

`mysample.ks`:
```
#version=RHEL8
%packages --instLangs="es_ES" --excludeWeakdeps
%end
```

`mytest.py`:
```python
import pykickstart.parser as ksparser

version = ksparser.version.makeVersion('RHEL8')
ks = ksparser.KickstartParser(version)
ks.readKickstart('mysample.ks')

ks.handler.url(url='install_tree_url')

print(str(ks.handler))
```

```bash
# python3 mytest.py > new.ks
# ksvalidator -v RHEL8 new.ks 

Checking kickstart file new.ks

The following problem occurred on line 10 of the kickstart file:

unrecognized arguments: --inst-langs=es_ES --exclude-weakdeps
```

**Comment by bcl** [comment]
There are 2 problems you have run into. The first is that the #version= line in the kickstart is just there for the user's benefit. pykickstart, anaconda, and koji do not use it at all -- despite there being a `versionFromFile` function available. So unless there is a way to tell koji to treat it as a RHEL8 kickstart it is going to default to using the highest supported version in the copy of pykickstart that is installed on the system running koji.

But you also found a bug :) Ends up when we aliased the --inst-langs to --instLangs the output function was changed to always output the new form. I've added a check for the version (Fedora 32) where the change was made, and that will be in the next release.
